src/qz/utils/ConnectionUtilities.java

```diff
@@ -0,0 +1,29 @@
+package qz.utils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+
+import qz.common.Constants;
+
+public final class ConnectionUtilities {
+    private ConnectionUtilities() {
+    }
+
+    public static InputStream getInputStream(String urlString) throws IOException {
+        URLConnection urlConn = new URL(urlString).openConnection();
+        urlConn.setRequestProperty("User-Agent", userAgent());
+        return urlConn.getInputStream();
+    }
+
+    public static String userAgent() {
```

Comment:
This should never change once the JVM has been loaded so it should just be a `private static` variable (which would eliminate the `return` problem).

If you want to keep it in a function for cleanliness purposes, you should still store it in a variable to avoid redundant `replaceAll` calculations, etc and the function should be private since it isn't used anywhere else.

e.g. 

```java
private static USER_AGENT = String.format("%s/%s (%s; %s) Java/%s (%s)",
    Constants.ABOUT_TITLE.replaceAll("[^a-zA-Z]", ""),
    Constants.VERSION,
    System.getProperty("os.name"),
    System.getProperty("os.version"),
    System.getProperty("java.version"),
    System.getProperty("java.vendor"));
```